### PR TITLE
LLMNR client support

### PR DESCRIPTION
**Pull request by jukkar**
Link-Local multicast name resolution (LLMNR) is a multicast DNS technique used in Windows networks. This PR adds support for LLMNR client functionality. What this means in practice, is that all the DNS queries (unless it is mDNS query) are sent to [ff02::1:3]:5355 or 224.0.0.252:5355 instead of unicast DNS server port 53.
This PR depends on mDNS PR #1341 so there are couple of extra patches here that belong to that mDNS PR.

### CODE REVIEW

**Comment by pfalcon** [comment]
I don't see Jira: mentioned in the commit, so a question: why do we need this? Why mDNS wouldn't be enough? (I'm not much familiar with either, so will need to do my homework before I'm able to review these.)

**Comment by jukkar** [comment]
> I don't see Jira: mentioned in the commit, so a question: why do we need this? Why mDNS wouldn't be enough? (I'm not much familiar with either, so will need to do my homework before I'm able to review these.)

I am not sure yet whether we need this, there is one use case but I will talk about that a bit later. This PR is not urgent and we can ignore it for time being. Anyway LLMNR was easy to implement so decided to send it for review.

**Comment by pfalcon** [comment]
> Anyway LLMNR was easy to implement so decided to send it for review.

Based on my homework-in-progress, as easy as using another multicast address and port number comparing to mDNS. Hail to vendor tug of war on the Internet standards (Apple vs Microsoft in this case).

**Comment by pfalcon** [comment]
This requires a rebase.

**Comment by jukkar** [comment]
> This requires a rebase.

Yes, I know. Currently this is bottom of the pile and won't be touched (at least by me) until more urgent things are done.

**Comment by pfalcon** [comment]
Sounds good, added "prio: low" label.

File: samples/net/dns_resolve/README.rst [hunk 1]
```diff
@@ -13,6 +13,14 @@ If a CNAME is received, the DNS resolver will create another DNS query.
 The number of additional queries is controlled by the
 DNS_RESOLVER_ADDITIONAL_QUERIES Kconfig variable.
 
+The multicast DNS (mDNS) client resolver support can be enabled by setting
+CONFIG_MDNS_RESOLVER Kconfig variable.
```
Comment (dbkinder) [inline samples/net/dns_resolve/README.rst]: you can generate a link reference to a Kconfig options by using ``:option:`CONFIG_MDNS_RESOLVER` ``

File: samples/net/dns_resolve/README.rst [hunk 2]
```diff
@@ -122,9 +106,15 @@ Open a terminal window and type:
 
     $ killall -s KILL dnsmasq
 
-
 Try to launch the dnsmasq application again.
 
+For testing mDNS, use Avahi script in net-tools project:
```
Comment (dbkinder) [inline samples/net/dns_resolve/README.rst]: For clarity,

     For testing mDNS, use the Avahi script, also found in the GitHub
     zephyrproject-rtos/net-tools project area:

**Review by dbkinder — CHANGES_REQUESTED** [review]
some suggested edits

**Comment by pfalcon** [comment]
@jukkar : This needs rebase/conflict resolve (if targeted for 1.10).

**Comment by jukkar** [comment]
Not going to make 1.10

**Comment by jukkar** [comment]
This has bitrotted and requires more testing thus closing it. I will re-submit after testing this better.

**Pull request closed without merging**